datalake/src/main/java/com/sequenceiq/datalake/service/upgrade/recovery/SdxUpgradeRecoveryService.java

```diff
@@ -51,6 +53,12 @@
     @Inject
     private WebApplicationExceptionMessageExtractor exceptionMessageExtractor;
 
+    @Inject
+    private Flow2Handler flow2Handler;
+
+    @Inject
+    private FlowChainLogService flowChainLogService;
+
     public SdxRecoveryResponse triggerRecovery(String userCrn, NameOrCrn clusterNameOrCrn, SdxRecoveryRequest recoverRequest) {
         SdxCluster cluster = sdxService.getByNameOrCrn(userCrn, clusterNameOrCrn);
```

Comment:
Should this not be validating the recovery before initiating it? You put code in the ResizeRecoveryService which validates before performing recovery, so I assumed it would be done here too.

Reply:
The selector service is responsible for validating before triggering. Neither trigger method needs to validate as it's already been done.